Sources/Bond/AppKit/NSOutlineView+ObservableTree.swift

```diff
@@ -0,0 +1,152 @@
+            case .delete(let at):
+                let parent = rootNode[at.dropLast()]
+                outlineView.removeItems(at: IndexSet(integer: at.item), inParent: parent, withAnimation: deleteAnimation ?? [])
+            case .update(let at):
+                let parent = rootNode[at.dropLast()]
+                let item = outlineView.child(at.item, ofItem: parent)
+                outlineView.reloadItem(item)
+            case .move(let from, let to):
+                let fromParent = rootNode[from.dropLast()]
+                let toParent = rootNode[to.dropLast()]
+                outlineView.moveItem(at: from.item, inParent: fromParent, to: to.item, inParent: toParent)
+            }
+        }
+
+        outlineView.endUpdates()
+    }
+}
+
+
+public extension SignalProtocol where Element: ObservableCollectionEventProtocol, Element.UnderlyingCollection: TreeNodeProtocol, Error == NoError {
+
+    @discardableResult
+    public func bind(to outlineView: NSOutlineView, animated: Bool = true, createCell: @escaping (Element.UnderlyingCollection, Element.UnderlyingCollection.Value?, NSTableColumn?, NSOutlineView) -> NSView?) -> Disposable {
+        let binder = OutlineViewBinder(measureCell: nil, createCell: createCell)
+        if !animated {
+            binder.deleteAnimation = nil
+            binder.insertAnimation = nil
+        }
+        return self.bind(to: outlineView, using: binder)
+    }
+
+    @discardableResult
+    public func bind(to outlineView: NSOutlineView, using binder: OutlineViewBinder<Element.UnderlyingCollection>) -> Disposable {
+        let dataSource = Property<Element.UnderlyingCollection?>(nil)
+        let disposable = CompositeDisposable()
+
+        disposable += outlineView.reactive.delegate.feed(
+            property: dataSource,
+            to: #selector(NSOutlineViewDelegate.outlineView(_:heightOfRowByItem:)),
+            map: { (dataSource: Element.UnderlyingCollection?, outlineView: NSOutlineView, item: Element.UnderlyingCollection.Value?) -> CGFloat in
+                guard let dataSource = dataSource else { return outlineView.rowHeight }
+                return binder.height(for: item, outlineView: outlineView, dataSource: dataSource) ?? outlineView.rowHeight
+            }
+        )
+
+        disposable += outlineView.reactive.delegate.feed(
+            property: dataSource,
+            to: #selector(NSOutlineViewDelegate.outlineView(_:viewFor:item:)),
+            map: { (dataSource: Element.UnderlyingCollection?, outlineView: NSOutlineView, tableColumn: NSTableColumn, item: Element.UnderlyingCollection.Value?) -> NSView? in
+                guard let dataSource = dataSource else { return nil }
+                return binder.cell(for: item, tableColumn: tableColumn, outlineView: outlineView, dataSource: dataSource)
+            }
+        )
+
+        disposable += outlineView.reactive.dataSource.feed(
```

Comment:
In theory, something like this should work I would expect:

```swift
disposable += outlineView.reactive.dataSource.feed(
    property: dataSource,
    to: #selector(NSOutlineViewDataSource.outlineView(_:child:ofItem:)),
    map: { (dataSource: Element.UnderlyingCollection?, _: NSOutlineView, child: Int, item: Element.UnderlyingCollection?) -> Any in
        guard let item = item else {
            let index = dataSource!.startIndex.advanced(by: child, atLevel: 0)
            return dataSource![index]
        }

        let index = item.startIndex.advanced(by: child, atLevel: 0)
        return item[index]
    }
)
```

Reply:
@tonyarnold what about following:

```swift
        disposable += outlineView.reactive.dataSource.feed(
            property: dataSource,
            to: #selector(NSOutlineViewDataSource.outlineView(_:child:ofItem:)),
            map: { (dataSource: Element.UnderlyingCollection?, _: NSOutlineView, child: Int, item: Element.UnderlyingCollection?) -> Any in
                guard let item = item else {
                    // if item is nil, then we return root node
                    return dataSource!.value
                }
                // otherwise return item's child
                return item.children[child]
            }
        )

        disposable += outlineView.reactive.dataSource.feed(
            property: dataSource,
            to: #selector(NSOutlineViewDataSource.outlineView(_:numberOfChildrenOfItem:)),
            map: { (dataSource: Element.UnderlyingCollection?, _: NSOutlineView, item: Element.UnderlyingCollection?) -> Int in
                return item?.children.count ?? 0
            }
        )
```

Reply:
@tonyarnold btw, you'll have to pull the latest changes. I added a constraint that TreeNodeProtocol.Children have to have `Index == Int`.

Reply:
Both of those suggestions work 👍